src/components/resource/resource-edit.js

```diff
@@ -1,49 +1,64 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 
+import { Icon } from '@folio/stripes-components';
+
 import ManagedResourceEdit from './edit-managed-title';
 import CustomResourceEdit from './edit-custom-title';
 
-export default function ResourceEdit({ model, ...props }) {
-  let initialValues = {};
-  let View;
-
-  if (model.isTitleCustom === true || model.destroy.params.isTitleCustom === true) {
-    View = CustomResourceEdit;
-    initialValues = {
-      isSelected: model.isSelected,
-      isVisible: !model.visibilityData.isHidden,
-      customCoverages: model.customCoverages,
-      coverageStatement: model.coverageStatement,
-      hasCoverageStatement: model.coverageStatement.length > 0 ? 'yes' : 'no',
-      customEmbargoValue: model.customEmbargoPeriod.embargoValue,
-      customEmbargoUnit: model.customEmbargoPeriod.embargoUnit,
-      customUrl: model.url,
-      proxyId: model.proxy.id
-    };
-  } else if (model.isTitleCustom === false) {
-    View = ManagedResourceEdit;
-    initialValues = {
-      isSelected: model.isSelected,
-      isVisible: !model.visibilityData.isHidden,
-      customCoverages: model.customCoverages,
-      coverageStatement: model.coverageStatement,
-      hasCoverageStatement: model.coverageStatement.length > 0 ? 'yes' : 'no',
-      customEmbargoValue: model.customEmbargoPeriod.embargoValue,
-      customEmbargoUnit: model.customEmbargoPeriod.embargoUnit,
-      proxyId: model.proxy.id
-    };
+export default class ResourceEdit extends Component {
+  static propTypes = {
+    model: PropTypes.object.isRequired
+  };
+
+  renderRequestErrorMessage() {
+    const { model } = this.props;
+
+    return (
+      <p data-test-eholdings-resource-edit-error>
+        {model.request.errors[0].title}
+      </p>
+    );
   }
 
-  return (
-    <View
-      model={model}
-      initialValues={initialValues}
-      {...props}
-    />
-  );
-}
+  indicateModelIsNotLoaded() {
+    const { model } = this.props;
+
+    return model.request.isRejected
+      ? this.renderRequestErrorMessage()
+      : (
+        <Icon
+          icon="spinner-ellipsis"
+          iconSize="small"
+        />
+      );
+  }
+
+  renderView() {
+    const {
+      model,
+      ...props
```

Comment:
You have only model in propTypes, so please check whethter you meed this ...props or need to add props to proptypes